*being updated by Narcis, 3rd May 2023*

# Generating Hi-C heatmaps

## Background


## Running Pretext on galaxy

### step 1: mapping Hi-C reads to the **scaffolded** genome

This step is similar to the scaffolding process; we use [**Map with BWA-MEM2**](https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy0) to map the Hi-C Illumina reads (Forward (R1) and reverse (R2)) to the genome, but we use the **scaffolded genome generated by YaHS** as the reference genome. Then filter and merge the mapped reads using [**Filter and merge** chimeric reads from Arima Genomics](https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy0). The resulting SAM/BAM file is input for Pretext. 


### step 2: run Pretext to generate the Hi-C heatmap
From Tools panel choose [**PretextMap converts SAM or BAM files into genome contact maps**](https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.8+galaxy2) with the following settings:

In **Input dataset in SAM or BAM format**: Output of filter and merge
**Sort by**: Don't sort
**How to filter**: No filter

and `Run Tool`

